data/transactions/logic/assembler.go

```diff
@@ -1580,14 +1580,36 @@ func getSpec(ops *OpStream, name string, args []string) (OpSpec, string, bool) {
 	}
 	spec, ok := OpsByName[ops.Version][name]
 	if !ok {
-		spec, ok = OpsByName[AssemblerMaxVersion][name]
+		var err error
+		spec, err = unknownOpcodeComplaint(name, ops.Version)
+		// unknownOpcodeComplaint's job is to return a nice error, so err != nil
+		ops.error(err)
+	}
+	return spec, spec.Name, ok
+}
+
+// unknownOpcodeComplaint returns the best error it can for a missing opcode,
+// plus a "standin" OpSpec, if possible.
+func unknownOpcodeComplaint(name string, v uint64) (OpSpec, error) {
```

Comment:
just a question, not suggesting a move: This method is also introduced for future reasoning about opcode deprecation, right?

Reply:
Yeah, I broke it out when doing deprecation. But I left it there as it seems handy, even though nothing is deprecated yet.